api/src/main/java/org/openmrs/api/handler/PatientDataUnvoidHandler.java

```diff
@@ -68,14 +68,19 @@ public void handle(Patient patient, User originalVoidingUser, Date origParentVoi
 			if (CollectionUtils.isNotEmpty(orders)) {
 				for (Order order : orders) {
 					if (order.isVoided() && order.getDateVoided().equals(origParentVoidedDate)
-					        && order.getVoidedBy().equals(originalVoidingUser)) {
+							&& order.getVoidedBy().equals(originalVoidingUser)) {
 						os.unvoidOrder(order);
 					}
 				}
 			}
-
+			
 			CohortService cs = Context.getCohortService();
-			cs.patientUnvoided(patient, patient.getVoidedBy(), patient.getDateVoided(), patient.getVoidReason());
+			if (patient.getId() != null) {
+				if (patient.getDateVoided().equals(origParentVoidedDate) &&
+						patient.getVoidedBy().equals(originalVoidingUser)) {
```

Comment:
Calling `Context.getPatientService().unvoidPatient(patient);`
gives error: `org.openmrs.api.ValidationException: 'Cohort id=2 name=Example Cohort size=2' failed to validate with reason: memberships: Cohort.patientAndMemberShouldBeVoided`
because of this in CohortValidator.java:
```
					if ((p.getVoided() && !member.getVoided())
							|| (member.getStartDate() == null || dateCompare == 1)) {
```

At this point, the patient is voided and the membership isn't. So my question, at what point the patient should be unvoided? Because at these stages: PatientDataUnvoidHandler.handle(), CohortServiceImpl.patientUnvoided(), the patient's voided fields are still set. Hence the validation fails in CohortValidator. I hope you get my question.